src/analytics/analytics-tracker-window.ts

```diff
@@ -1,9 +1,67 @@
+import { isAnalyticsTrackerWindow } from './is-analytics-step-tracker-window';
+
 export interface AnalyticsTracker {
     track(step: string, data: any): void;
-    hit(hitName: string, data: any): void;
-    hasPayloadLimit(data: any): boolean;
 }
 
 export default interface AnalyticsTrackerWindow extends Window {
     analytics: AnalyticsTracker;
 }
+
+interface AnalyticsTrackerWindowGA extends AnalyticsTrackerWindow {
+    ga(command: string, eventName: string, payload: any): void;
+}
+
+function isAnalyticsTrackerWindowGA(window: Window | AnalyticsTrackerWindowGA): window is AnalyticsTrackerWindowGA {
+    return window && 'ga' in window && typeof window.ga === 'function';
+}
+
+export const isGoogleAnalyticsAvailable = () => {
+    return isAnalyticsTrackerWindow(window) && isAnalyticsTrackerWindowGA(window);
+};
+
+export const sendGoogleAnalytics = (type: string, payload: any) => {
+    if (isAnalyticsTrackerWindowGA(window)) {
+        window.ga('send', type, {
+            ...payload,
+            nonInteraction: false,
+        });
+    }
+};
+
+/**
+ * Max size of the payload for the Google Analytics module
+ * if the limit will be succeeded, the GA throwing a silent error,
+ * and only in debug mode you can see it
+ */
+export const ANALYTICS_MAX_URI_LENGTH = 8096;
+
+interface AnalyticPayload {
+    [key: string]: string | number | AnalyticPayload;
+}
+
+export const isPayloadSizeLimitReached = (obj: AnalyticPayload) => {
+    return serializeAnalyticsEventPayload(obj).length >= ANALYTICS_MAX_URI_LENGTH;
+};
+
+const serializeAnalyticsEventPayload = (obj: AnalyticPayload): string => {
+    return Object.keys(obj).reduce((acc: string[], key) => {
+        const type = typeof obj[key];
+
+        if (type === 'string' || type === 'number') {
+            return [
+                ...acc,
+                `${key}=${obj[key]}`,
+            ];
+        }
+
+        if (type === 'object' && obj[key] !== null) {
+            return [
+                ...acc,
+                serializeAnalyticsEventPayload(obj[key] as AnalyticPayload),
```

Comment:
@davidchin We have added `as AnalyticPayload` because TS doesn't understand that `number` and `string` cases are already handled.

Let us know if there is a better solution 🙇